sql/src/main/java/io/crate/planner/optimizer/Optimizer.java

```diff
@@ -41,22 +42,25 @@
 
     private final List<Rule<?>> rules;
     private final Supplier<Version> minNodeVersionInCluster;
+    private final Functions functions;
 
-    public Optimizer(List<Rule<?>> rules, Supplier<Version> minNodeVersionInCluster) {
-        this.rules = rules;
+    public Optimizer(Functions functions, Supplier<Version> minNodeVersionInCluster, List<Class<? extends Rule<?>>> rulesToInclude) {
+        this.rules = loadRules(rulesToInclude);
         this.minNodeVersionInCluster = minNodeVersionInCluster;
+        this.functions = functions;
     }
 
     public LogicalPlan optimize(LogicalPlan plan, TableStats tableStats, TransactionContext txnCtx) {
-        LogicalPlan optimizedRoot = tryApplyRules(plan, tableStats, txnCtx);
+        LogicalPlan optimizedRoot = tryApplyRules(rules, plan, tableStats, txnCtx);
```

Comment:
```suggestion
        LogicalPlan optimizedRoot = tryApplyRules(plan, tableStats, txnCtx);
```

There is still a `rules` attribute.